CMakeLists.txt

```diff
@@ -1,10 +1,14 @@
 cmake_minimum_required (VERSION 3.2)
+set(CMAKE_CXX_STANDARD 14)
```

Comment:
I put this once in root instead of under `examples` and `tests`, since it applies not only to them, but to the actual library as well